src/anemoi/utils/commands/metadata.py

```diff
@@ -0,0 +1,398 @@
+# (C) Copyright 2024 Anemoi contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+import json
+import logging
+import os
+import shutil
+import subprocess
+from argparse import ArgumentParser
+from argparse import Namespace
+from tempfile import TemporaryDirectory
+from typing import Any
+from typing import Dict
+
+import yaml
+
+from . import Command
+
+LOG = logging.getLogger(__name__)
+
+EDITOR_OPTIONS = {"code": ["--wait"]}
+
+
+class Metadata(Command):
+    """Edit, remove, dump or load metadata from a checkpoint file."""
+
+    def add_arguments(self, command_parser: ArgumentParser) -> None:
+        """Add command line arguments to the parser.
+
+        Parameters
+        ----------
+        command_parser : ArgumentParser
+            The argument parser to which the arguments will be added.
+        """
+        from anemoi.utils.checkpoints import DEFAULT_NAME
+
+        command_parser.add_argument("path", help="Path to the checkpoint.")
+
+        group = command_parser.add_mutually_exclusive_group(required=False)
+        remove_metdata_group = command_parser.add_argument_group(title="Metadata removal")
```

Comment:
Is the remove group needed? In fact, should `--remove` not be added to the mutually exclusive group? Since it is mutually exclusive with the other commands. `--inplace` can then just be added to the parser, and the output argument can also be reused. 

It would also be useful to have `--edit` be not-inplace by default and then have `--inplace` as an option. But I'm happy to address that later, but for now you could already add `--inplace` to the general parser.